src/super_gradients/training/utils/detection_utils.py

```diff
@@ -664,30 +663,42 @@ class DetectionCollateFN:
     """
 
     def __call__(self, data) -> Tuple[torch.Tensor, torch.Tensor]:
-        batch = default_collate(data)
-        ims, targets = batch[0:2]
-        return ims, self._format_targets(targets)
+        images_batch, labels_batch = list(zip(*data))
```

Comment:
`images_batch, labels_batch = list(zip(*data))` will only work if each sample of data is a tuple of 2 elements. It's true that we should have 2, but if someone uses for instance a dataset with crowd_target=True, and eventually uses `DetectionCollateFN` then the code would crash (there might be other scenarios)

I think the way to do would be something like this:
```python
def __call__(self, data) -> Tuple[torch.Tensor, torch.Tensor]:
    batch = default_collate(data)
    if len(batch) != 2:
         raise RuntimeError(f"`DetectionCollateFN` only supports Datasets that return a tuple (image, target), but got a tuple of len={len(batch)}")
    ims, targets = batch[0:2]
    return self._format_images(images_batch), self._format_targets(labels_batch)
```
What do you think ?

Reply:
I would argue that crashing is a desired behavior for such usage. Why would we want a collate that silently ignores items from a loader? This is by the way what happened in CI of 41674fb, I fixed collates in the test

Reply:
Crashing due to `ims, targets = batch[0:2]` would not be informative for people who don't know SG, this is why I propose to make it explicit:
``` python
    if len(batch) != 2:
         raise RuntimeError(f"`DetectionCollateFN` only supports Datasets that return a tuple (image, target), but got a tuple of len={len(batch)}")
```

Reply:
++ for more user friendly error messages.
Yes, that's classic - change one place, and get +10 more improvement requests.

Reply:
`if len(batch) != 2` is wrong, it's a batch length, not a sample length. Added an expcetion not to drag the message over all collates, see 452fb19
Using `try except` for [“zero-cost” exceptions](https://docs.python.org/3.11/whatsnew/3.11.html#misc), don't want any extra work on each iteration